cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -87,6 +85,7 @@
 #include "arrow/util/optional.h"
 #include "arrow/util/task_group.h"
 #include "arrow/util/thread_pool.h"
+#include "arrow/util/windows_fixup.h"
```

Comment:
I don't know if moving this line is deliberate but it _might_ break Windows builds (or it might no; I honestly have no idea :-)).

Reply:
I undid this change (over-zealous IDE)